lib/log.c

```diff
@@ -157,14 +158,39 @@ _cs_matches_filter_(struct qb_log_callsite *cs,
 	return match;
 }
 
+/**
+ * @internal
+ * @brief Format a log message into a string buffer
+ *
+ * @param[out] str  Destination buffer to contain formatted string
+ * @param[in]  cs   Callsite containing format to use
+ * @param[in]  ap   Variable arguments for format
+ */
+static inline int32_t
+cs_format(char *str, struct qb_log_callsite *cs, va_list ap)
+{
+	va_list ap_copy;
+	int len;
+
+	va_copy(ap_copy, ap);
+	len = vsnprintf(str, QB_LOG_MAX_LEN, cs->format, ap_copy);
```

Comment:
I'm pretty sure the qb_vsnprintf functions in lib/log_format.c should be used here.  i don't remember the history, but I believe those qb format functions are faster and provide greater cross compatibility.